shared_model/interfaces/iroha_internal/transaction_batch_template_definitions.hpp

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_TRANSACTION_BATCH_TEMPLATE_DEFINITIONS_HPP
+#define IROHA_TRANSACTION_BATCH_TEMPLATE_DEFINITIONS_HPP
+
+#include "interfaces/iroha_internal/transaction_batch.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    namespace {
+      /**
+       * check if all transactions belong to the same batch
+       * @param txs transactions to be checked
+       * @return true if all transactions from the same batch and false
+       * otherwise
+       */
+      bool allTxsInSameBatch(const types::SharedTxsCollectionType &txs) {
```

Comment:
Is it possible to move the definition to source file?

Reply:
I think, it's better to leave it here, in anonymous namespace, so it will be seen and used only by this transaction unit and no-one else

Reply:
Yes, but the code will be duplicated for different translation units, isn't it?